src/output/mod.rs

```diff
@@ -236,19 +250,37 @@ pub fn metrics(config: Config) {
             let snapshot = SnapshotterBuilder::new().build().snapshot();
 
             let buf = match format {
-                MetricsFormat::Json => serde_json::to_vec(&snapshot).expect("failed to serialize"),
-                MetricsFormat::Messagepack => {
-                    rmp_serde::encode::to_vec(&snapshot).expect("failed to serialize")
+                MetricsFormat::Json => Snapshot::to_json(&snapshot).expect("failed to serialize"),
+                MetricsFormat::MsgPack | MetricsFormat::Parquet => {
+                    Snapshot::to_msgpack(&snapshot).expect("failed to serialize")
                 }
             };
             let _ = writer.write_all(&buf);
 
-            // Write newline for ndjson
-            if format == MetricsFormat::Json {
-                let _ = writer.write_all(b"\n");
-            }
-
             next += Duration::from_secs(1);
         }
     }
+    let _ = writer.flush();
+    drop(writer);
+
+    // Post-process metrics into a parquet file
+    if format == MetricsFormat::Parquet {
+        let tmpfile = file.1;
```

Comment:
You might want to use tempfile crate? https://docs.rs/tempfile/latest/tempfile/

Reply:
I'd originally written it with `tempfile::NamedTempFile`, but I got worried about the case of running on shared hardware and having some job cleaning up `/tmp` outside our control.

Reply:
Hmm, I missed that there was a `new_in` method: https://docs.rs/tempfile/latest/tempfile/struct.NamedTempFile.html#method.new_in

Probably best of both worlds with that...